doc/jointcal.tex

```diff
@@ -29,527 +43,603 @@
+
+The computation of the Jacobian and the gradient is performed in the
+respective \ClName{PhotometryFit} and \ClName{AstrometryFit} classes.
+The methods \RoutineName{leastSquareDerivativesMeasurement} and
+\RoutineName{leastSquareDerivativesReference} compute the contributions
+to the Jacobian and gradient of the $\chi^{2}$ from the measurement
+terms and the references terms respectively. In these routines, the
+Jacobian is represented as a list of triplets $(i,j,J_{ij})$ describing
+its elements. This list is then transformed into a representation
+of sparse matrices suitable for algebra, and in particular suitable
+to evaluate the product $J^{T}J$. Once we have evaluated $H\equiv J^{T}J$,
+we can solve eq. \ref{eq:hessian_step} using a Cholesky factorization.
+For sparse linear algebra, the Cholmod and Eigen packages provide
+the required functionality. It turns out that for the mappings we
+have currently employed, the calculation of $J^{T}J$ and the factorization
+are the most CPU intensive parts of the calculations, and there is
+hence not much to be gained in speeding up the calculation of derivatives.
+For the factorization, we have tried both Eigen and Cholmod (via the
+Eigen interface) and their speeds differ by less than 10\%.
+
+\subsection{Photometry example}
+
+As an illustrative example, we will work through a particular photometry
+mapping, consisting of a constant zero-point per CCD ($f_{0}$: the
+CCD filter response) and an $(n+m)$th order 2-D Chebyshev polynomial
+($\sum a_{j,k}T_{j}(u)T_{k}(v)$: the optics+sky response, where $(u(x,y),v(x,y))$
+are the focal plane coordinates of pixel $(x,y)$ on a given CCD)
+per visit. Thus, the mapping will be
+\[
+M_{\gamma i}(\eta,S_{\gamma i})=M_{CCD}(f_{0}^{-1},f_{\gamma i})M_{visit}(a_{j,k},x_{\gamma i},y_{\gamma i})=f_{\gamma i}[f_{0}]^{-1}\sum_{j=0}^{j=n}\sum_{k=0}^{k=m}a_{j,k}T_{j}(u_{\gamma i})T_{k}(v_{\gamma i})=\phi_{\gamma i}
+\]
+where we will fit $f_{0}^{-1}$ instead of $f_{0}$ in order to simplify
+the derivatives (with respect to $\eta=\left(f_{0}^{-1},a_{j,k}\forall j,k\right)$).
+Computing those derivatives gives us:
+\begin{eqnarray*}
+\nabla D_{\gamma i} & = & (\frac{\partial D_{\gamma i}}{\partial f_{0}^{-1}},\frac{\partial D_{\gamma i}}{\partial a_{0,0}},\ldots,\frac{\partial D_{\gamma i}}{\partial a_{n,m}},\frac{\partial D_{\gamma i}}{\partial F_{i}})
+\end{eqnarray*}
+where, for the measurement terms we have (recall eq. \ref{eq:photometry_residual_vector_measurement}),
+\begin{eqnarray*}
+\frac{\partial D_{\gamma i}}{\partial f_{0}^{-1}} & = & f_{\gamma i}M_{visit}(a_{j,k}x_{\gamma i},y_{\gamma i})\\
+\frac{\partial D_{\gamma i}}{\partial a_{j,k}} & = & f_{\gamma i}f_{0}^{-1}T_{jk}(u_{\gamma i},v_{\gamma i})\\
+\frac{\partial D_{\gamma i}}{\partial F_{i}} & = & -1
+\end{eqnarray*}
+and for the reference terms (recall eq. \ref{eq:photometry_residual_vector_reference}),
+\[
+\nabla D_{j}=\frac{\partial D_{j}}{\partial F_{j}}=1
+\]
+
+\subsection{Astrometry example}
```

Comment:
I am confused--this section does not look like an Astrometry example.

Reply:
Yeah, that was a typo, that I fixed after I'd sent this to you to review. It's Photometry.